internal/transport/http2_client.go

```diff
@@ -137,11 +137,23 @@ type http2Client struct {
 	connectionID uint64
 }
 
-func dial(ctx context.Context, fn func(context.Context, string) (net.Conn, error), addr string) (net.Conn, error) {
-	if fn != nil {
-		return fn(ctx, addr)
+func dial(ctx context.Context, fn func(context.Context, string) (net.Conn, error), addr resolver.Address, useProxy bool, grpcUA string) (net.Conn, error) {
+	if fn == nil {
+		fn = func(fCtx context.Context, fAddr string) (net.Conn, error) {
+			if networkType := addr.Attributes.Value("network_type"); networkType != nil {
```

Comment:
Read `network_type` outside of this closure. It should help simplify things a bit.

Reply:
Made this change.